cmd/earthly/app/run.go

```diff
@@ -157,6 +160,18 @@ func (app *EarthlyApp) run(ctx context.Context, args []string) int {
 			"Error: No SetFatalError called appropriately. This should never happen.")
 	}()
 
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	userKilled := atomic.Bool{}
+
+	go func() {
+		for range sigChan {
+			userKilled.Store(true)
```

Comment:
There could be a race condition, if this doesn't execute quickly enough.

Maybe it's better to store this here https://github.com/earthly/earthly/blob/main/cmd/earthly/main.go#L61 instead. It would also deduplicate the need for `signal.Notify`.

Reply:
Updated. The signal is now provided to this code in a thread-safe way. 

I removed the warning line as it's somewhat duplicated, wasn't working, & is writing a log message after everything is closed. Ok, to remove?

We already have these messages.

```
^CReceived signal interrupt. Cleaning up before exiting...
Canceled
View logs at https://cloud.earthly.dev/builds/821b3352-5a00-42b0-aa3f-73e835367da2
```